src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,78 @@
+package duke;
+
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+import duke.commands.Command;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.tasks.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Represents the initialization of Duke application.
+ * Initializes the application and starts the interaction with the user.
+ */
+public class Duke {
+    private TaskList tasks;
+    private Ui ui;
+    private Storage storage;
+
+    /**
+     * @param filePath to read/write task list data
+     * @throws IOException
+     */
+    public Duke(String filePath) throws IOException {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Runs the main Duke program.
+     *
+     */
+    public void run() {
+        boolean isActive = true;
+
+        String welcomeMsg = "Hey, I'm duke.Duke.\n"
+                + "What's up?\n";
+        String exitMsg = "Bye! Hope I helped!\n"
+                + "See you next time :)\n";
+
+        System.out.println(welcomeMsg);
+        System.out.println("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
```

Comment:
It may be useful to abstract out these messages as class constants.

Something like this for example:
```java
private static final String MSG_WELCOME = "Hey, I'm duke.Duke.\nWhat's up?\n";
private static final String MSG_EXIT = "Bye! Hope I helped!\nSee you next time :)\n";
```

Reply:
Good suggestion, thank you!